azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/method_call.py

```diff
@@ -0,0 +1,39 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a class representing method calls that are received and responded to.
+"""
```

Comment:
I'll probably combine this into a singular module with Message, called data_structures.py or something in the final version. Essentially, a module for user-facing objects.

Reply:
I don't mind having smaller modules. one potentially desirable consequence would be the ability to cleanly isolate features into modules. Maybe for modularity purposes, maybe if there's such a thing as tree shaking in python? if this is all internal and has no bearing on the public interface though, we can always refactor later. your call.

---
In reply to: [265780208](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r265780208) [](ancestors = 265780208)

Reply:
These objects I'm talking about would all be user-facing, however, they wouldn't be exposed through this module, they'd be exposed through the package itself, so as far as how we organize them, that's just an internal implementation detail.

AFAIK there's no tree shaking tool or process for Python, at least, not that I've heard of.

---
In reply to: [266185461](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266185461) [](ancestors = 266185461,265780208)

Reply:
UPDATE: Instead of combining them into the same module, they've been combined into the same package. This helps with organization by not forcing them to be a part of the client layer, which was awkward because they were used in the transport as well (resulting in a somewhat circular-ish import)

---
In reply to: [266596563](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266596563) [](ancestors = 266596563,266185461,265780208)